src/cluster_asm.c

```diff
@@ -272,6 +272,55 @@ size_t asmGetPeakSyncBufferSize(void) {
     return peak;
 }
 
+static inline int asmIsSlotImporting(void) {
+    if (!asmManager || listLength(asmManager->tasks) == 0) return 0;
+
+    /* Only support a single task at a time now, so only check the first task */
+    asmTask *task = listNodeValue(listFirst(asmManager->tasks));
+    /* We only check the destination side, the source side `pauseActions` will
+     * pause the write traffic (including expire/evict). */
+    /* TODO: for failed state, is it safe to expire? cleanup may take long time */
```

Comment:
> TODO: for failed state, is it safe to expire? cleanup may take long time */

You mean on the destination side? We can't expire keys as we don't own them, no?

Reply:
removed